Result/ResultType.swift

```diff
@@ -64,6 +64,30 @@ public extension ResultType {
 	}
 }
 
+/// Protocol used to constrain a version of `map` for `transform`s that throw.
+public protocol ErrorTypeConvertible: ErrorType {
+	typealias ConvertibleType = Self
+	static func errorFromErrorType(error:ErrorType) -> ConvertibleType
+}
+
+public extension ResultType where Error: ErrorTypeConvertible {
+
+	/// Returns the result of applying `transform` to `Success`es’ values, or wrapping thrown errors.
+	public func map<U>(@noescape transform: Value throws -> U) -> Result<U, Error> {
+		return flatMap { value in
+			do {
+				return .Success(try transform(value))
+			}
+			catch {
+				guard let convertedError = Error.errorFromErrorType(error) as? Error else {
+					fatalError("Unable to convert error type '\(error.dynamicType)' to type '\(Error.self)'")
+				}
+				return .Failure(convertedError)
```

Comment:
I don't think so. The return type of the `transform` is quite different, and part of the reason for this PR is to be able to pass in functions that are not aware of `Result`.